File: admin/app/Models/SermonParagraph.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class SermonParagraph extends Model
{
    use HasFactory;

    protected $primaryKey = 'paragraph_id';

    protected $fillable = [
        'sermon_id',
        'paragraph_no',
        'text',
        'start_time_ms',
        'end_time_ms',
    ];

    protected function casts(): array
    {
        return [
            'paragraph_no' => 'integer',
            'start_time_ms' => 'integer',
            'end_time_ms' => 'integer',
        ];
    }

    public function sermon(): BelongsTo
    {
        return $this->belongsTo(Sermon::class, 'sermon_id', 'sermon_id');
    }
}

File: lib/models/sermon_paragraph.dart
class SermonParagraph {
  const SermonParagraph({
    required this.id,
    required this.sermonId,
    required this.number,
    required this.sortOrder,
    required this.text,
    this.startMs,
    this.endMs,
  });

  final int id;
  final int sermonId;
  final String number;
  final int sortOrder;
  final String text;
  final int? startMs;
  final int? endMs;

  factory SermonParagraph.fromMap(Map<String, Object?> map) {
    return SermonParagraph(
      id: map['paragraph_id'] as int,
      sermonId: map['sermon_id'] as int,
      number: map['paragraph_no']?.toString() ?? '',
      sortOrder: map['paragraph_no'] as int? ?? 0,
      text: map['text']?.toString() ?? '',
      startMs: map['start_time_ms'] as int?,
      endMs: map['end_time_ms'] as int?,
    );
  }
}
